week4.tex

```diff
@@ -0,0 +1,182 @@
+\documentclass[letterpaper]{article}
+
+\input{setup}
+\input{defns}
+
+\title{15-791 ATPL \\ Week 4 Notes}
+\author{Kyle Booker and Alex Knox}
+\date{\today}
+\begin{document}
+\maketitle
+
+\section{System F}
+
+We can extend a basic language like the simply typed lambda calculus (STLC) with variable types. The addition of universally quantified types gives rise to the following language, called System F. It can also be more descriptively called the polymorphic $\lambda$-calculus or the second-order $\lambda$-calculus.
+
+$$
+A \coloncolonequals X \mid \ansTy* \mid \arrTy*{A_1}{A_2} \mid \allTy*{X}{A}
+$$
+$$
+M \coloncolonequals x \mid \lamEx*{x}{M} \mid \appEx*{M_1}{M_2} \mid \tlamEx*{X}{M} \mid \tapEx*{M}{A}
+$$
+
+Note that $x$ refers to \textit{term variables}, whereas $X$ refers to \textit{type variables}.
+
+System F mostly copies the statics and dynamics from the STLC, with the addition of some extra rules for incorporating type variables. Those are replicated below:
+
+Static rules relating to quantifiers:
+
+$$
+\defrule[Lam][sta:lam]
+  {\Gamma \entails[\Delta, X]{\isOfTp{M}{A}}}
+  {\Gamma \entails[\Delta]{\isOfTp{\tlamEx*{X}{M}}{\allTy*{X}{M}}}}
+$$
+
+$$
+\defrule[Lam-App][sta:app]
+  {\Gamma \entails[\Delta]{\isOfTp{M}{\allTy*{X}{A}}} \\
+   \Delta \entails{B \text{ type}}}
+  {\Gamma \entails[\Delta]{\isOfTp{\tapEx*{M}{B}}{\Sub{B}{X}{A}}}}
+$$
+
+Dynamic rules relating to quantifiers:
+$$
+\defrule[Lam][dyn:lam]
+  {M \stepsTo M'}
+  {\tlamEx*{X}{M} \stepsTo{} \tlamEx*{X}{M'}}
```

Comment:
I believe `\Lambda` needs to be final. See PFPL 16.3b.

Reply:
Oh, very true, nice catch!